src/ShowPage/ShowPage.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import { BrowserRouter, Route, Link } from 'react-router-dom';
+import { getSingleMovie } from '../apiCalls'
+import './ShowPage.css';
+
+
+
+
+// let allGenres;
+
+class ShowPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentMovie: {},
+      error: ''
+    }
+  }
+
+  componentDidMount() {
+    this.determineSingleMovie(this.props.id)
+  }
+
+
+  determineSingleMovie = (id) => {
+    getSingleMovie(id)
+    .then(singleMovie => this.setState({currentMovie: singleMovie}))
+    .catch(error => this.setState({ error: error }));
+  }
+
+    getGenres = () => {
+      return this.props.genres.map(genre => {
+        return <p>genre.name</p>
+      })
+    }
+
+    render() {
```

Comment:
It might be worth destructuring your props, to make your render more readable! Like:
`const { title, backdropPath, tagline, runTime, averageRating, userRating, overview, releaseDate, budget, revenue } = this.props;`

And then in the render, you can just put `{title}` instead of `{this.props.title}`, etc.